Hyper/Meshes/Chunk.cs

```diff
@@ -98,36 +109,49 @@ public bool Build(Vector3 location, float deltaTime, float brushWeight, int radi
                     if (DistanceSquared(x, y, z, xi, yi, zi) <= radius * radius)
                     {
                         _voxels[xi, yi, zi].Value -= deltaTime * brushWeight * Gaussian(xi, yi, zi, x, y, z, 0.1f);
+                        flag = true;
                     }
                 }
             }
         }
 
-        UpdateMesh();
+        if(flag) UpdateMesh();
 
         var error = GL.GetError();
         if (error != ErrorCode.NoError) Logger.Error(error);
 
-        return true;
+        return flag;
     }
 
     public void UpdateCollisionSurface(Simulation simulation, BufferPool bufferPool)
     {
+        if (collisionSurface == null)
+        {
+            CreateCollisionSurface(simulation, bufferPool);
+            return;
+        }
         simulation.Shapes.RemoveAndDispose(_shape, bufferPool);
         var mesh = MeshHelper.CreateMeshFromChunk(this, bufferPool);
         _shape = simulation.Shapes.Add(mesh);
         simulation.Statics[_handle].SetShape(_shape);
     }
+    public void DisposeCollisionSurface(Simulation simulation, BufferPool bufferPool)
+    {        
+        simulation.Statics.Remove(_handle);
+        simulation.Shapes.RemoveAndDispose(_shape, bufferPool); //datarace if in other thread
```

Comment:
do these data races occur? If so, we need to fix that

Reply:
I don't know if that's related or not, but when mining/building I'm getting `InvalidOperation` messages in the console. This needs to be investigated

Reply:
When building we can fall through the map. I think this should be solved in this PR

Reply:
Basically i don't know how to remove a CollisionSurface... if i do it in a way provided above, the simulation often tries to use it resulting in a null exception. Right now these surfaces stay indefinitely, which is obviously an issue.

Reply:
I do not think falling through the map is an issue on my part, or issue at all. I dramatically increased the volume and strength of building/mining. Because of that, when we mine/build, a large surface of terrain shifts its position in a single tick and sometimes the surface "teleports" on the other side of the character. Then we fall...